### ISSUE TITLE
Improve `wai-aria` rule for `input` element with role `switch`

### ISSUE THREAD

**Issue by kodai3** [markuplint/markuplint#1517]
- Markuplint version: `v4.3.0`
- Parser lang: HTML or JSX and more...
- Node.js version: `v18.19.0`
- OS: macOS

## Describe the bug
When I implement `Switch` component with `input` element with role `switch` according to https://www.w3.org/WAI/ARIA/apg/patterns/switch/examples/switch-checkbox/

markuplint throw error  
`Require the "aria-checked" ARIA state on the "switch" role`

then when I add the `aria-checked` field it throws differently like below
`The "aria-checked" ARIA state must not use on the "input" element. Add the "checked" attribute if you use the ARIA state`

## Code Example

```html
<input
    type="checkbox"
    role="switch"
    checked={isChecked}
    {...otherProps}
  />
```

## The current behavior
Follow linter suggestion throws another error and we are not able to fix.

## The expected behavior
`Require the "aria-checked" ARIA state on the "switch" role` should not be thrown.
related to https://github.com/w3c/html-aria/issues/227
